user-service/src/component-tests/java/uk/gov/gchq/palisade/component/user/service/UserServiceAsyncProxyErrorTest.java

```diff
@@ -0,0 +1,149 @@
+    };
+    private final Function<Integer, UserRequest> noUserRequestFactoryObj = i -> {
+        try {
+            return this.mapper.treeToValue(noUserRequestFactoryNode.apply(i), UserRequest.class);
+        } catch (JsonProcessingException e) {
+            throw new SerializationFailedException("Failed to convert contract test data to objects", e);
+        }
+    };
+
+    @Test
+    void testContextLoads() {
+        assertThat(userServiceAsyncProxy).isNotNull();
+    }
+
+    @Test
+    void testGetUserSuccess() {
+        // Given a user request
+        final UserRequest userRequest = requestFactoryObj.apply(1);
+        final User user = new User().userId("test-user-id");
+        // When adding to the cache
+        this.userServiceAsyncProxy.addUser(user);
+
+        // Then retrieving from the cache
+        final CompletableFuture<AuditableUserResponse> subject = this.userServiceAsyncProxy.getUser(userRequest);
+
+        // Check the CompletableFuture hasn't finished
+        assertThat(subject.isDone()).isFalse();
+        // Then complete the future
+        AuditableUserResponse auditableUserResponse = subject.join();
+        // Then Check it has been completed
+        assertThat(subject.isDone()).isTrue();
+
+        // Then the service suppresses exception and populates Audit object
+        assertThat(auditableUserResponse.getAuditErrorMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isNull();
+
+        // Then the cached User is the same as the original User
+        assertThat(auditableUserResponse.getUserResponse().getUserId())
+                .isEqualTo(user.getUserId().getId());
+    }
+
+    @Test
+    void testGetUserFailure() {
+        // When user has been added
+
+        // Then retrieving a different user
+        final UserRequest userRequest = noUserRequestFactoryObj.apply(1);
+        final CompletableFuture<AuditableUserResponse> subject = this.userServiceAsyncProxy.getUser(userRequest);
+
+        // Check the CompletableFuture hasn't finished
+        assertThat(subject.isDone()).isFalse();
+        // Then complete the future
+        AuditableUserResponse auditableUserResponse = subject.join();
+        // Then Check it has been completed
+        assertThat(subject.isDone()).isTrue();
+
+        // Then check that there is an error message
+        assertThat(auditableUserResponse.getAuditErrorMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isNotNull();
+
+        // Then check the error message contains the correct message
+        assertThat(auditableUserResponse.getAuditErrorMessage().getError().getMessage())
+                .as("verify that exception is propagated into an auditable object and returned")
+                .isEqualTo(NoSuchUserIdException.class.getName() + ": No userId matching Not-a-real-user found in cache");
+    }
```

Comment:
I'd prefer to check the `assertThat(error).hasType(NoSuchUserIdException.class)` rather than inspecting the `getMessage` string

Reply:
The issue with this is that the "Throwable" error is actually a `CompletionException` with a message that contains the NoSuchUserIdException class.
Hence why we have the classname and the message in the `isEqualTo` check.